test/index.ts

```diff
@@ -112,6 +127,38 @@ export function defaultPropTestParams<T = unknown>(): fc.Parameters<T> {
     return seed !== undefined ? { ...baseParams, seed, path } : baseParams;
 }
 
+/**
+ * Wraps a fast-check property test assertion and provides a helpful error message if the test fails.
+ *
+ * @param assertion - The fast-check property test assertion to run
+ * @throws Error with reproduction command and original error message if the assertion fails
+ *
+ * @remarks
+ * The error message includes:
+ * 1. A command that can be used to reproduce the exact failing test case
+ * 2. The original error message from fast-check
+ */
+export async function runAndProvideReproduction(
```

Comment:
Let's look into using a global error handler to implement this functionality instead.